main.go

```diff
@@ -12,8 +14,37 @@ type ProcessSettings struct {
 	FlushAfter int
 }
 
+// ReceivedMessage is the message as received by a transport backend.
+type ReceivedMessage struct {
+	Payload          []byte
+	Attributes       map[string]string
+	ProviderMetadata interface{}
+}
+
+// QueueIdentifer provides a function to get the identifier for a backend to sub to a queue
+// structs satisfying FirehoseBackend should have FirehoseSubscriptions filled with QueueIdentifers
+type QueueIdentifer interface {
+	GetIdentifer() string
+}
+
 // FirehoseBackend is used for consuming messages from a transport and read/write to storage
 type FirehoseBackend interface {
+	// Receive messages from configured queue(s) and provide it through the channel. This should run indefinitely
+	// until the context is canceled. Provider metadata should include all info necessary to ack/nack a message.
+	// The channel must not be closed by the backend.
+	Receive(ctx context.Context, queueIdentifier QueueIdentifer, numMessages uint32, visibilityTimeout time.Duration, messageCh chan<- ReceivedMessage) error
```

Comment:
This is changed to take in a `QueueIdentifier` since Im guessing the follower will only be reading from one queue (based on the image in the doc)